test/src/test/java/org/corfudb/runtime/collections/CorfuStoreSecondaryIndexTest.java

```diff
@@ -104,6 +105,80 @@ public void testSecondaryIndexes() throws Exception {
         }
     }
 
+    /**
+     * Test secondary indexes set on Enum type
+     */
+    @Test
+    public void testSecondaryIndexesOnEnum() throws Exception {
+        // Get a Corfu Runtime instance.
+        CorfuRuntime corfuRuntime = getTestRuntime();
+
+        // Creating Corfu Store using a connected corfu client.
+        CorfuStoreShim shimStore = new CorfuStoreShim(corfuRuntime);
+
+        // Define a namespace & table name
+        final String someNamespace = "some-namespace";
+        final String tableName = "EnumTable";
+        final String[] activities = new String[] {"basket", "football", "soccer"};
+
+        // Create & Register the table, this table has an Enum field marked as secondary key
+        Table<UuidMsg, ActivitySchedule, ManagedMetadata> activitiesTable = shimStore.openTable(
+                someNamespace,
+                tableName,
+                UuidMsg.class,
+                ActivitySchedule.class,
+                ManagedMetadata.class,
+                TableOptions.builder().build());
+
+        // Populate table with X records
+        final int numRecords = 42;
+        final int daysInAWeek = 7;
+        final ManagedMetadata metadata = ManagedMetadata.newBuilder().setCreateUser("user_UT").build();
+
+        for (int i = 0; i < numRecords; i++) {
+            UUID uuid = UUID.randomUUID();
+            UuidMsg key = UuidMsg.newBuilder()
+                    .setMsb(uuid.getMostSignificantBits()).setLsb(uuid.getLeastSignificantBits())
+                    .build();
+
+            try (ManagedTxnContext txn = shimStore.tx(someNamespace)) {
+                ActivitySchedule activity = ActivitySchedule.newBuilder()
```

Comment:
nit - move this construction out of the tx context?

Reply:
no need, its not used anywhere else, it's the object within the transaction scope.